lib/kubernetes-deploy/kubernetes_resource/deployment.rb

```diff
@@ -123,5 +126,35 @@ def find_latest_rs(deployment_data)
       rs.sync(latest_rs_data)
       rs
     end
+
+    def min_unavailable_rollout
+      @definition.dig('metadata', 'annotations', 'kubernetes-deploy.shopify.io/required-rollout')
+    end
+
+    def minimum_unavailable_replicas_to_succeeded
+      desired = @desired_replicas
+
+      case min_unavailable_rollout
+      when 'maxUnavailable'
+        max_unavailable = @definition.dig('spec', 'strategy', 'rollingUpdate', 'maxUnavailable')
```

Comment:
We need to use the value from `sync` rather than `@definition`, possibly with a fallback to `@definition` (like we do for `@progress_deadline`), since this value can be server-side defaulted.